src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -159,6 +159,35 @@ def test_get_unique_name(self, cephadm_module):
         new_mgr = cephadm_module.get_unique_name('mgr', 'myhost', existing)
         match_glob(new_mgr, 'myhost.*')
 
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm", _run_cephadm('[]'))
+    def test_valid_url(self, cephadm_module):
+        # Test with valid URLs
+        test_cases = [
+            ("192.168.100.100:9090", "prometheus multi-cluster targets updated"),
+            ("127.0.0.1:8080", "prometheus multi-cluster targets updated"),
+            ("192.168.100.100", "prometheus multi-cluster targets updated"),
+        ]
+        with with_host(cephadm_module, 'test'):
+            with with_service(cephadm_module, ServiceSpec(service_type='prometheus'), CephadmOrchestrator.apply_prometheus, 'test'):
+                for url, expected_output in test_cases:
+                    c = cephadm_module.set_prometheus_target(url)
+                    assert wait(cephadm_module,
+                                c) == expected_output
+
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm", _run_cephadm('[]'))
+    def test_invalid_url(self, cephadm_module):
+        # Test with invalid URLs
+        test_cases = [
+            ("http://example.com:9090", "Invalid service endpoint 'http://example.com:9090'. It should only be in the format host_ip:port."),
+            ("192.168.100.100:99999", "Invalid service endpoint. Port out of range 0-65535")
```

Comment:
how is this UT is passing when the message in the code is `The port must be between 1 and 65535.` ?

Reply:
Updated